cli/root.go

```diff
@@ -331,3 +365,32 @@ func FormatCobraError(err error, cmd *cobra.Command) string {
 	helpErrMsg := fmt.Sprintf("Run '%s --help' for usage.", cmd.CommandPath())
 	return cliui.Styles.Error.Render(err.Error() + "\n" + helpErrMsg)
 }
+
+func checkVersions(cmd *cobra.Command, client *codersdk.Client) error {
+	flag := cmd.Flag("no-version-warning")
+	if suppress, _ := strconv.ParseBool(flag.Value.String()); suppress {
+		return nil
+	}
+
+	clientVersion := buildinfo.Version()
+
+	info, err := client.BuildInfo(cmd.Context())
+	if err != nil {
+		return xerrors.Errorf("build info: %w", err)
+	}
+
+	fmtWarningText := `client/server versions do not match
+client version: %s
+server version: %s
+to download the appropriate version run 'curl -L https://coder.com/install.sh | sh -s -- --version %s'
```

Comment:
I feel like we could shorten this up a bit. What do you think about:
```
version mismatch: client %s, server %s
download the server version with: ...
```

Reply:
:100: